core/src/main/java/org/apache/calcite/rel/rules/ReduceExpressionsRule.java

```diff
@@ -738,6 +740,20 @@ protected static boolean reduceExpressionsInternal(RelNode rel,
     // Replace predicates on CASE to CASE on predicates.
     boolean changed = new CaseShuttle().mutate(expList);
 
+    // distinct for IN rexNode
+    for (RexNode rexNode : predicates.pulledUpPredicates) {
```

Comment:
I think changing this VALUE here complicates the issue. Maybe it would be better to look into the "substituteSubQuery" option, before converting it to OR or RexSubquery.

Reply:
Yes，I agree，would do some changes in future.